### PR TITLE
Migrate from intltool to gettext

### PR DESCRIPTION
**Pull request by yetist**
When the gettext version is 0.20 (on Archlinux), it will fail on some po files, I am not sure if this is a bug in gettext 0.20 or a bug in po file.

### CODE REVIEW

**Comment by raveit65** [comment]
Archlinux build with gettext-0.20 failed at another place.
https://travis-ci.org/mate-desktop/mate-themes/jobs/580040772#L1189
```
/usr/sbin/msgfmt: error while opening "an.1po" for reading: No such file or directory
```

**Comment by raveit65** [comment]
It seems to work fine and all translation strings are in mate-themes.pot.
But do you have a idea why this line in mate-themes.pot is wrong?
`"Content-Type: text/plain; charset=CHARSET\n"`
Should be
`"Content-Type: text/plain; charset=UTF-8\n"`
Issue still exits in master , so it isn't related to PR, but it is annoying because it breaks `distcheck`if it goes into *.po files.
And we have to fix it by hand every time we send the resource file to transifex.

File: configure.ac [hunk 1]
```diff
@@ -10,10 +10,10 @@ AM_INIT_AUTOMAKE([1.9 tar-ustar dist-xz no-dist-gzip check-news])
...
 
-IT_PROG_INTLTOOL([0.35.0])
 PKG_PROG_PKG_CONFIG([0.19])
+AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT([external])
 
-AM_GLIB_GNU_GETTEXT
 GETTEXT_PACKAGE=AC_PACKAGE_NAME
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package.])
```
Comment (rbuj) [inline configure.ac]: You can remove these lines:
```
GETTEXT_PACKAGE=AC_PACKAGE_NAME
AC_SUBST(GETTEXT_PACKAGE)
AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package.])
```

**Comment by yetist** [comment]
> Installation with `sudo make install` doesn't install any locale.
> It seems that that all locale will be installed first, but later they will be removed.
> 
Yes, it's ok, please see the $top_srcdir/Makefile.am.
```
# we don't want to install mo files, all translations are already stored
# in theme files
install-data-local:
	$(MAKE) -C po uninstall
```

**Comment by rbuj** [comment]
You shouldn't rename index.theme to index.desktop, all files are Desktop files in po/POTFILES.in:
```patch
diff --git a/po/Makevars b/po/Makevars
index 5c3836c5..66ae5e12 100644
--- a/po/Makevars
+++ b/po/Makevars
@@ -8,7 +8,7 @@ subdir = po
 top_builddir = ..
 
 # These options get passed to xgettext.
-XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ --keyword=C_:1c,2 --keyword=NC_:1c,2 --keyword=g_dngettext:2,3 --add-comments
+XGETTEXT_OPTIONS = --from-code=UTF-8 --language=Desktop --add-comments
 
 # This is the copyright holder that gets inserted into the header of the
 # $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
```

**Pull request merged**